### ISSUE TITLE
Embed Prolog or lisp for a rules base

### ISSUE THREAD

**Issue by truedat101** [IoTone/ProjectMagNET#28]
Add a prolog or lisp to generate a rules base that can handle the more complex stuff behind #23 .  It is understood that things like the NPM package manager use prolog for their rules on package dependency management.  This is just an idea.

**Comment by pedroalvesbatista** [comment]
Would this be some sort of DSL here ?

**Comment by truedat101** [comment]
Yes.  Rules engines are sort of built into prolog for sure, and prolog variants are rich enough to actually implement lisp inside of it.

**Comment by truedat101** [comment]
RISC-V engine is best idea for embedded processor.
